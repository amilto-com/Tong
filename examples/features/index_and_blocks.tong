// Demonstrate indexing expressions and explicit block / anonymous fn blocks

let xs = [10,20,30]
print("idx0", xs[0])
print("idx1", xs[1])

// nested indexing on an expression
let m = [[1,2],[3,4]]
print("m_1_0", m[1][0])

// anonymous function with block body capturing local bindings
let make = fn a b {
  let c = a + b
  let d = c * 2
  d + 1  // implicit last expr value
}
print("make", make(2,3))

// explicit block expression used inline (should yield last expression value)
let val = {
  let t = 5
  let u = t * 3
  u + 7
}
print("blockVal", val)

// chaining after block expression
let blockIdx = { [100,200,300] }[1]
print("blockIdx", blockIdx)
