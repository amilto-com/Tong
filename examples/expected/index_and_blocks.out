idx0 10 idx1 20 m_1_0 3 make 11 blockVal 22 blockIdx 200